src/clients/c++/perf_client/perf_client.cc

```diff
@@ -1303,9 +1304,20 @@ main(int argc, char** argv)
 
       for (PerfStatus& status : summary) {
         uint64_t avg_queue_ns = status.server_stats.queue_time_ns /
-                                status.server_stats.request_count;
-        uint64_t avg_compute_ns = status.server_stats.compute_time_ns /
-                                  status.server_stats.request_count;
+                                status.server_stats.execution_count;
```

Comment:
Have to be careful with counts. The success count tells you how many successful requests there were. inference_count tells you how many inferences were performed where a batch size n request is counted as n inferences. execution_count tells you how many times the model ran... which with dynamic batching can be significantly less than the number of requests or inference_count. I'm not sure you are using execution_count correctly here and maybe elsewhere.

Reply:
Is it safe to assume the success metric count to be same as other metric counts(compute_*, queue*) as we were doing earlier? Or should we just record the average using there own individual metric counts?

Reply:
I was wrong assuming the execution count to be equal to the number of times these metrics were added. Fixed.